domain/application/service/unit_test.go

```diff
@@ -207,3 +212,102 @@ func (s *unitServiceSuite) TestUpdateCAASUnitNotAlive(c *gc.C) {
 	err := s.service.UpdateCAASUnit(context.Background(), coreunit.Name("foo/666"), UpdateCAASUnitParams{})
 	c.Assert(err, jc.ErrorIs, applicationerrors.ApplicationNotAlive)
 }
+
+// TestSetReportedUnitAgentVersionInvalid is here to assert that if pass a
+// junk agent binary version to [Service.SetReportedUnitAgentVersion] we get
+// back an error that satisfies [coreerrors.NotValid].
+func (s *unitServiceSuite) TestSetReportedUnitAgentVersionInvalid(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	err := s.service.SetReportedUnitAgentVersion(
+		context.Background(),
+		coreunit.Name("0"),
+		coreagentbinary.Version{
+			Number: version.Zero,
+		},
+	)
+	c.Check(err, jc.ErrorIs, coreerrors.NotValid)
+}
+
+// TestSetReportedUnitAgentVersionSuccess asserts that if we try to set the
```

Comment:
Hi I think this should be TestSetReportedUnitAgentVersionNotFound instead of TestSetReportedUnitAgentVersionSuccess

Reply:
fixed.